lib/spack/docs/packaging_guide.rst

```diff
@@ -3453,6 +3441,13 @@ directory, including cached virtual indices.
 
 To remove all of the above, the command can be called with ``--all``.
 
+When called with positional arguments, cleans up temporary files only
+for a particular package, by deleting the
+expanded/checked out source code *and* any downloaded archive.  If
```

Comment:
Just to clarify, does `spack clean foo` delete the stage and the cached download archive? Or just the former?